### PR TITLE
Plugins: Refactor secrets plugin .proto file

### LINKED ISSUE #50456: Update .proto message definition according to feedback
https://github.com/grafana/grafana/pull/49544#discussion_r892575488

https://github.com/grafana/grafana/pull/49544#discussion_r892579693

https://github.com/grafana/grafana/pull/49544#discussion_r892583639

https://github.com/grafana/grafana/pull/50542#discussion_r895562488

- Rename message types to use <verb><subject>Request/Response convention
- Create custom response type for each endpoint instead of using SecretsErrorResponse for everything (allows for easier mods later)
- Consider renaming "RemoteSecretsManager" (remote is implied)
- Make error field into "user friendly error" so that we can show detailed and controlled error messages in the front end

### PR DESCRIPTION
**Pull request by mmandrus**
**What this PR does / why we need it**:
This PR addresses non-blocking comments in [some prior PRs ](https://github.com/grafana/grafana/issues/50456)about the .proto message definition for grpc secrets management 

- Rename `RemoteSecretsManager` to just `SecretsManager`
- Follow <verb><subject>Request/Response convention
- Give each grpc endpoint its own message type for greater flexibility
- Remote `error` from response messages and add `userFriendlyError` to support [better error handling](https://github.com/grafana/grafana/pull/50542#discussion_r895562488)

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/50456

### CODE REVIEW

**Review by guicaulada — APPROVED** [review]

**Comment by guicaulada** [comment]
I find `UserFriendlyError` a little too discriptive, I don't see why we can't just use `Error` and specify that it should be something user friendly. But maybe it's better to be overly discriptive than not.

**Comment by mmandrus** [comment]
> I find `UserFriendlyError` a little too discriptive, I don't see why we can't just use `Error` and specify that it should be something user friendly. But maybe it's better to be overly discriptive than not.

IMO the problem with just `Error` is that a grpc method can either return an actual error _or_ a message payload, not both, so it's unclear why you'd ever use a response payload with `Error` populated. The more descriptive name queues devs on its use.

File: pkg/plugins/backendplugin/secretsmanagerplugin/secretsmanager.proto [hunk 1]
```diff
@@ -3,54 +3,62 @@ package secretsmanagerplugin;
...
-    rpc Set(SecretsSetRequest) returns (SecretsErrorResponse);
-    rpc Del(SecretsDelRequest) returns (SecretsErrorResponse);
-    rpc Keys(SecretsKeysRequest) returns (SecretsKeysResponse);
-    rpc Rename(SecretsRenameRequest) returns (SecretsErrorResponse);
+service SecretsManager {
+    rpc Get(GetSecretRequest) returns (GetSecretResponse);
+    rpc Set(SetSecretRequest) returns (SetSecretResponse);
+    rpc Del(DelSecretRequest) returns (DelSecretResponse);
+    rpc Keys(ListSecretsRequest) returns (ListSecretsResponse);
+    rpc Rename(RenameSecretRequest) returns (RenameSecretResponse);
```
Comment (marefr) [inline pkg/plugins/backendplugin/secretsmanagerplugin/secretsmanager.proto:63]: What do you think of these suggestions?

```suggestion
    rpc GetSecret(GetSecretRequest) returns (GetSecretResponse);
    rpc SetSecret(SetSecretRequest) returns (SetSecretResponse);
    rpc DeleteSecret(DeleteSecretRequest) returns (DeleteSecretResponse);
    rpc ListSecrets(ListSecretsRequest) returns (ListSecretsResponse);
    rpc RenameSecret(RenameSecretRequest) returns (RenameSecretResponse);
```

**Review by marefr — APPROVED** [review]
LGTM definitely an improvement :+1: Added a suggestion

File: pkg/plugins/backendplugin/secretsmanagerplugin/secretsmanager.proto (hunk 1 above)
Comment (mmandrus) [reply pkg/plugins/backendplugin/secretsmanagerplugin/secretsmanager.proto:63]: I'm a big fan of descriptive variable and function names, so I like these. I will make the changes and merge. Thanks!

**Comment by mmandrus** [comment]
> Maybe worth mentioning @mmandrus - is that we're currently working on ways in the SDK to pass query error status back from a plugin to Grafana. So far we're thinking that we would use error codes as part of the proto response, and Grafana can then dictate what actual message to return to the user based on that code. We're still toying with ideas, but one is we will also probably have a place for metadata to go alongside the status code. Maybe something for us to think about together when things mature on this side

@wbrowne sounds great! Please loop me in when the time is right.

**Pull request merged**